src/main/java/com/revature/services/PersonLogicImpl.java

```diff
@@ -1,34 +1,68 @@
 package com.revature.services;
 
-import java.util.List;
+
+
+
+import java.util.ArrayList;
+
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Isolation;
 import org.springframework.transaction.annotation.Transactional;
 
 import com.revature.domain.Person;
+
 import com.revature.repositories.PersonRepository;
 
+
 @Service
 @Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
 public class PersonLogicImpl implements PersonLogic {
 
+
 	@Autowired
-	private PersonRepository personRepository;
+	private PersonRepository dao;
 	
 	@Override
-	public List<Person> getPersonByFirstName(String firstName) {
+	public ArrayList<Person> getPersonByFirstName(String firstName) {
 		
-		return personRepository.findByFirstName(firstName);
+		return  dao.findByFirstName(firstName);
 
 	}
 
 	@Override
-	public List<Person> getPersonByLastName(String lastName) {
+	public ArrayList<Person> getPersonByLastName(String lastName) {
 		
-		return personRepository.findByLastName(lastName);
+		return dao.findByLastName(lastName);
 	}
 	
 
+
+	@Override
+	public ArrayList<Person> getPersonsByPersonRole(int personRole) {
+		
+		return dao.findByPersonRole(personRole);
+	
+	}
+
+	@Override
+	public ArrayList<Person> getAllTrainees() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1319)